hikari/impl/cache.py

```diff
@@ -1295,12 +1289,13 @@ def clear_voice_states_for_channel(
         if not guild_record or not guild_record.voice_states:
             return cache_utility.EmptyCacheView()
 
-        cached_voice_states = {}
+        cached_voice_states = dict(
+            filter(lambda item: item[1].channel_id == channel_id, guild_record.voice_states.items())
+        )
```

Comment:
This seems really unnecessary, you can just copy the guild's voice stats before iterating over them like everywhere else in the cache stores

Reply:
Changed :+1: